store/sqlstore/store_test.go

```diff
@@ -4,22 +4,110 @@
 package sqlstore
 
 import (
+	"flag"
+	"os"
+	"sync"
 	"testing"
 
+	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/store"
+	"github.com/mattermost/mattermost-server/store/storetest"
 	"github.com/mattermost/mattermost-server/utils"
 )
 
-var sqlStore store.Store
+var storeTypes = []*struct {
+	Name      string
+	Func      func() (*storetest.RunningContainer, *model.SqlSettings, error)
+	Container *storetest.RunningContainer
+	Store     store.Store
+}{
+	{
+		Name: "MySQL",
+		Func: storetest.NewMySQLContainer,
+	},
+	{
+		Name: "PostgreSQL",
+		Func: storetest.NewPostgreSQLContainer,
+	},
+}
 
 func StoreTest(t *testing.T, f func(*testing.T, store.Store)) {
-	if sqlStore == nil {
-		utils.TranslationsPreInit()
-		utils.LoadConfig("config.json")
-		utils.InitTranslations(utils.Cfg.LocalizationSettings)
-		sqlStore = store.NewLayeredStore(NewSqlSupplier(nil), nil, nil)
+	defer func() {
+		if err := recover(); err != nil {
+			tearDownStores()
+			panic(err)
```

Comment:
Being super cautious about tearing down the stores. I *really* don't want to leave any containers behind no matter what.